lib/fluent/plugin/out_sumologic.rb

```diff
@@ -37,6 +37,10 @@ def request_headers(source_host, source_category, source_name, data_type, metric
       else
         raise RuntimeError, "Invalid #{metric_data_format}, must be graphite or carbon2 or prometheus"
       end
+
+      unless metadata.nil?
+        headers['X-Sumo-Metadata'] = metadata
```

Comment:
As discussed on Slack, let's confirm Metadata is the best use, as opposed to dimensions.

Reply:
I change it to dimensions, but regarding another discussion I'm not sure if we want to merge those changes or not